ark/ops/ops_all_gather_test.cc

```diff
@@ -68,7 +68,7 @@ void test_all_gather_8gpus_internal_msll(size_t nelem, int iter) {
             auto outputs = m.local_all_gather_msll(data, gpu_id, num_gpus);
             auto result =
                 ark::op_test("all_gather", m, {data}, {outputs},
-                             baseline_all_gather_msll<ark::half_t, 8>,
+                             baseline_all_gather_msll<ark::half_t, num_gpus>,
```

Comment:
Some old gcc versions cannot compile constexpr as a template argument.. we'd better avoid this